### PR TITLE
feat: accessibility tests

### PR DESCRIPTION
**Pull request by melniiv**
## Description
- Installed and configured with a11y plugin for storybook which allows to run the AXE tests automatically in the UI when storybook is running and run from scripts
- configured scripts
- Fixed found accessibility issues

### Closes

**[HCRC-123](https://helsinkisolutionoffice.atlassian.net/browse/HCRC-123):**

## Additional notes

[HCRC-123]: https://helsinkisolutionoffice.atlassian.net/browse/HCRC-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by karisal-anders** [comment]
Tried with node 20.18.0 on Windows 11 with "yarn" + "yarn storybook", and then when storybook was running, "yarn test-storybook" in another terminal. Needed to install web browsers for testing first though, so "yarn playwright install" and then "yarn test-storybook" went through. All tests passed:
<details>

<summary>All tests passed (Click to expand)</summary>

```
$ yarn test-storybook
yarn run v1.22.22
$ test-storybook
[test-storybook] The "preRender" hook is deprecated and will be removed in later versions. Please use "preVisit" instead in your test-runner config file.
[test-storybook] The "postRender" hook is deprecated and will be removed in later versions. Please use "postVisit" instead in your test-runner config file.
jest-haste-map: duplicate manual mock found: articles.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts

jest-haste-map: duplicate manual mock found: pageChildrenSearch.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\pageChildrenSearch.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\pageChildrenSearch.mock.d.ts

jest-haste-map: duplicate manual mock found: categories.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\categories.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\categories.mock.d.ts

jest-haste-map: duplicate manual mock found: pages.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\pages.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\pages.mock.d.ts

jest-haste-map: duplicate manual mock found: card.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\card\__mocks__\card.mock.d.ts
    * <rootDir>\dist\cjs\src\core\card\__mocks__\card.mock.d.ts

jest-haste-map: duplicate manual mock found: navigationMenu.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\navigation\__mocks__\navigationMenu.mock.d.ts
    * <rootDir>\dist\cjs\src\core\navigation\__mocks__\navigationMenu.mock.d.ts

jest-haste-map: duplicate manual mock found: navigationLanguages.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\navigation\__mocks__\navigationLanguages.mock.d.ts
    * <rootDir>\dist\cjs\src\core\navigation\__mocks__\navigationLanguages.mock.d.ts

jest-haste-map: duplicate manual mock found: navigationUniversalBarMenu.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\navigation\__mocks__\navigationUniversalBarMenu.mock.d.ts
    * <rootDir>\dist\cjs\src\core\navigation\__mocks__\navigationUniversalBarMenu.mock.d.ts

jest-haste-map: duplicate manual mock found: article.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\pageContent\__mocks__\article.mock.d.ts
    * <rootDir>\dist\cjs\src\core\pageContent\__mocks__\article.mock.d.ts

jest-haste-map: duplicate manual mock found: pageWithDiverseContent.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\pageContent\__mocks__\pageWithDiverseContent.mock.d.ts
    * <rootDir>\dist\cjs\src\core\pageContent\__mocks__\pageWithDiverseContent.mock.d.ts

jest-haste-map: duplicate manual mock found: page.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\pageContent\__mocks__\page.mock.d.ts
    * <rootDir>\dist\cjs\src\core\pageContent\__mocks__\page.mock.d.ts

jest-haste-map: duplicate manual mock found: notification.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\notification\__mocks__\notification.mock.d.ts
    * <rootDir>\dist\cjs\src\core\notification\__mocks__\notification.mock.d.ts

 PASS   browser: chromium  src/core/pageModules/SocialMediaFeedModule/SocialMediaFeedModule.stories.tsx (9.762 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageModules/ImageGalleryModule/ImageGalleryModule.stories.tsx (9.763 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageModules/StepsModule/StepsModule.stories.tsx (9.854 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/link/Link.stories.tsx (9.855 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/cardsList/CardsList.stories.tsx (9.968 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/common/components/htmlToReact/HtmlToReact.stories.tsx (10.39 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageModules/CardsModule/CardsModule.stories.tsx (13.044 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/carousel/Carousel.stories.tsx (13.546 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/page/Page.stories.tsx (13.72 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/apollo/apollo-custom-template.stories.tsx (13.83 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/collection/Collection.stories.tsx (14.35 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/archiveSearchPage/ArchiveSearchPage.stories.tsx (14.757 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/apollo/apollo.stories.tsx (14.803 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/card/Card.stories.tsx (14.915 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/common/components/grid/Grid.stories.tsx
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/common/components/text/Text.stories.tsx (5.217 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/linkBox/Link.stories.tsx (5.308 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageModules/ContentModule/ContentModule.stories.tsx (5.438 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageModules/CardModule/CardModule.stories.tsx (5.551 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/card/LargeCard.stories.tsx (5.51 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/contentContainer/ContentContainer.stories.tsx
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/common/components/list/List.stories.tsx
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageSection/PageSection.stories.tsx
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/pageContent/PageContent.stories.tsx (16.166 s)
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/core/hero/Hero.stories.tsx
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

 PASS   browser: chromium  src/common/components/tag/Tag.stories.tsx
  ● Console

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)

    console.log
      No accessibility violations detected!

      at DefaultTerminalReporter.<anonymous> (node_modules/axe-playwright/dist/reporter/defaultTerminalReporter.js:47:41)


Test Suites: 26 passed, 26 total
Tests:       49 passed, 49 total
Snapshots:   0 total
Time:        21.324 s
Ran all test suites.
Done in 22.75s.
```

</details>

There were some warnings given by `test-storybook`:
```
[test-storybook] The "preRender" hook is deprecated and will be removed in later versions. Please use "preVisit" instead in your test-runner config file.
[test-storybook] The "postRender" hook is deprecated and will be removed in later versions. Please use "postVisit" instead in your test-runner config file.
jest-haste-map: duplicate manual mock found: articles.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts
...
```
See the "All tests passed (Click to expand)" for all the warnings.

File: .storybook/test-runner.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+const { injectAxe, checkA11y } = require('axe-playwright');
+module.exports = {
+  preRender: async (page) => {
+    // before the story has been rendered inject AXE
+    await injectAxe(page);
+  },
+  postRender: async (page) => {
+    // after the story has been rendered rn AXE checksonly on root not the whole page
```
Comment (karisal-anders) [inline .storybook/test-runner.js:8]:
```suggestion
    // after the story has been rendered, run AXE checks only on root, not the whole page
```

File: public/mockServiceWorker.js [hunk 2]
```diff
@@ -8,111 +8,111 @@
  * - Please do NOT serve this file on production.
  */
 
-const INTEGRITY_CHECKSUM = '3d6b9f06410d179a7f7404d4bf4c3c70'
-const activeClientIds = new Set()
+const INTEGRITY_CHECKSUM = '3d6b9f06410d179a7f7404d4bf4c3c70';
+const activeClientIds = new Set();
```
Comment (karisal-anders) [inline public/mockServiceWorker.js:13]: This file starts with:
```javascript
/* eslint-disable */
/* tslint:disable */
/**
 * Mock Service Worker (1.2.3).
 * @see https://github.com/mswjs/msw
 * - Please do NOT modify this file.
 * - Please do NOT serve this file on production.
 */
```

This file should probably be completely ignored by prettier also so it doesn't try to change anything in this file.

File: .storybook/test-runner.js (hunk 1 above)
Comment (melniiv) [reply .storybook/test-runner.js]: fixed

File: public/mockServiceWorker.js (hunk 2 above)
Comment (melniiv) [reply public/mockServiceWorker.js]: fixed

**Comment by melniiv** [comment]
fixed the jest warnings

**Review by karisal-anders — APPROVED** [review]
Looks ok otherwise, only the jest-haste-map warnings still at the beginning of "test-storybook"'s output i.e.
```
$ yarn test-storybook
yarn run v1.22.22
$ test-storybook
jest-haste-map: duplicate manual mock found: articles.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts

jest-haste-map: duplicate manual mock found: categories.mock.d
  The following files share their name; please delete one of them:
    * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\categories.mock.d.ts
    * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\categories.mock.d.ts
...
```

**Comment by karisal-anders** [comment]
> Looks ok otherwise, only the jest-haste-map warnings still at the beginning of "test-storybook"'s output i.e.
> 
> ```
> $ yarn test-storybook
> yarn run v1.22.22
> $ test-storybook
> jest-haste-map: duplicate manual mock found: articles.mock.d
>   The following files share their name; please delete one of them:
>     * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts
>     * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\articles.mock.d.ts
> 
> jest-haste-map: duplicate manual mock found: categories.mock.d
>   The following files share their name; please delete one of them:
>     * <rootDir>\dist\src\core\archiveSearchPageContent\__mocks__\categories.mock.d.ts
>     * <rootDir>\dist\cjs\src\core\archiveSearchPageContent\__mocks__\categories.mock.d.ts
> ...
> ```

"yarn build" + "yarn storybook" in one terminal, then "yarn test-storybook" -> jest-haste-map warnings went away. So basically "yarn build" helped with removing the jest-haste-map warnings when running the storybook tests.

**Pull request merged**